src/kubectlutils.ts

```diff
@@ -182,25 +151,17 @@ export async function getKubernetesVersion(): Promise<string | undefined> {
 }
 
 export async function getNamedPodsFromKubectl(podNameRoot : string): Promise<string[]> {
-	return new Promise <string[]> ( async (resolve, reject) => {
-		await getPodsFromKubectlCli()
-		.then( (allPods) => {
-			let podArray = parseShellResult(allPods);
-			let outArray : string[] = [];
-			podArray.forEach(podName => {
-				if (podName.startsWith(podNameRoot)) {
-					outArray.push(podName);
-				}
-			});
-			if (outArray && outArray.length > 0) {
-				resolve(outArray);
-			} else {
-				resolve(undefined);
+	let outArray : string[] = [];
+	try {
+		const allPods: string = await getPodsFromKubectlCli();
+		let podArray = parseShellResult(allPods);
+		for (const podName of podArray) {
+			if (podName.startsWith(podNameRoot)) {
+				outArray.push(podName);
 			}
-			return;
-		}).catch ( (error) => {
-			reject(error);
-			return;
-		});
-	});
+		}
+	} catch (error) {
+		console.error(error);
```

Comment:
shouldn't we throw an error?

Reply:
yep

Reply:
I agree - similar situation to the one in getNamedListFromKubernetesThenParseList